examples/nyxstone-cli.cpp

```diff
@@ -1,176 +1,234 @@
-#include <boost/algorithm/hex.hpp>
-#include <boost/program_options.hpp>
-#include <boost/program_options/options_description.hpp>
-
-#include <algorithm>
 #include <iomanip>
 #include <iostream>
 #include <optional>
 
+#include <argh.h>
+
 #include "nyxstone.h"
 #include "tl/expected.hpp"
 
-namespace po = boost::program_options;
-
 using nyxstone::Nyxstone;
 using nyxstone::NyxstoneBuilder;
 
 void print_bytes(const std::vector<uint8_t>& bytes);
 std::optional<std::vector<Nyxstone::LabelDefinition>> parse_labels(std::string_view labelstr);
 void print_instructions(const std::vector<Nyxstone::Instruction>& instructions);
 
-int main(int argc, char** argv)
+std::vector<uint8_t> hex_decode(std::string const& hex_string)
 {
-    // clang-format off
-    po::options_description desc("Allowed options");
-    desc.add_options()
-        ("help", "Show this message")
-        ("arch", po::value<std::string>()->default_value("x86_64"),
-            "LLVM triple or architecture identifier of triple. For the most common architectures, we recommend:\n"
-            "x86_32: `i686-linux-gnu`\n"
-            "x86_64: `x86_64-linux-gnu`\n"
-            "armv6m: `armv6m-none-eabi`\n"
-            "armv7m: `armv7m-none-eabi`\n"
-            "armv8m: `armv8m.main-none-eabi`\n"
-            "aarch64: `aarch64-linux-gnueabihf`\n"
-            "Using shorthand identifiers like `arm` can lead to Nyxstone not being able to assemble certain instructions."
-        )
-        ("cpu", po::value<std::string>()->default_value(""),
-            "LLVM cpu specifier, refer to `llc -mtriple=ARCH -mcpu=help` for a comprehensive list")
-        ("features", po::value<std::string>()->default_value(""),
-            "LLVM features to enable/disable, comma seperated feature strings prepended by '+' or '-' to "
-            "enable or disable respectively. Refer to `llc -mtriple=ARCH -mattr=help` for a comprehensive list")
-        ("address", po::value<std::string>()->default_value("0"), "Address")
-    ;
-
-    po::options_description desc_asm("Assembling");
-    desc_asm.add_options()
-        ("labels", po::value<std::string>(), "Labels, for example \"label0=0x10,label1=0x20\"")
-        ("assemble,A", po::value<std::string>(), "Assembly")
-    ;
-
-    po::options_description desc_disasm("Disassembling");
-    desc_disasm.add_options()
-        ("disassemble,D", po::value<std::string>(), "Byte code in hex, for example: \"0203\"")
-    ;
-    // clang-format on
-
-    desc.add(desc_asm);
-    desc.add(desc_disasm);
-
-    po::variables_map varmap;
-    try {
-        po::store(po::parse_command_line(argc, argv, desc), varmap);
-    } catch (const std::exception& e) {
-        std::cerr << "Error occured: " << e.what() << "\n";
-        return 2;
+    auto input_size = hex_string.size();
+    if (input_size % 2 != 0) {
```

Comment:
I think it is a bit unclear here (to a first time reader) why we return nothing if the `hex_string` does not hold an even number of characters. Having a comment along the lines of `This function parses instruction bytes, which we expect to always be typed out in full` or renaming the function to something like `decode_instruction_bytes` might make this clearer.